src/bitcoin-cli.cpp

```diff
@@ -235,7 +235,21 @@ UniValue CallRPC(const std::string& strMethod, const UniValue& params)
     assert(output_buffer);
     evbuffer_add(output_buffer, strRequest.data(), strRequest.size());
 
-    int r = evhttp_make_request(evcon.get(), req.get(), EVHTTP_REQ_POST, "/");
+    // check if we should use a special wallet endpoint
+    std::string endpoint = "/";
+    std::string walletName = GetArg("-wallet", "");
```

Comment:
In commit "Add urlencode/decode via libevent2"

Commit description should be updated to reflect that this is adding a new option to bitcoin-cl. Or maybe this commit should solely update bitcoin-cli, and the urlencode should be moved to the "Select wallet based on the given endpoint" commit.

Also the new bitcoin-cli option needs to be documented, see https://github.com/bitcoin/bitcoin/pull/10650#discussion_r123489861